src/launcher.cpp

```diff
@@ -16,12 +16,19 @@
     along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
+#include <QCoreApplication>
 #include "launcher.h"
 
 Launcher::Launcher(QObject *parent) :
     QObject(parent),
     m_process(new QProcess(this))
 {
+    m_plasmaWelcomeInstalled = programExists("plasma-welcome");
```

Comment:
Please check for $DESKTOP_SESSION in this case to not exec plasma-welcome on GNOME and vice versa (think of folks who try to install all install options). We should check for gnome-tour as well.

Reply:
Otherwise I love the idea. As this tool would still provide fallback for DEs without a dedicated greeter.